[mongo-c-driver] build failure

**Host Environment**
- OS: Windows 8.1
- Compiler: MSVC (from Microsoft Visual Studio 2019)

**To Reproduce**
Steps to reproduce the behavior:
vcpkg should be located in a directory with a long path like `C:\Users\Denis\AppData\Local\pmm\1.4.2\vcpkg-790910f79f653978f90aadd958abf3c407215552`.

`./vcpkg install mongo-c-driver`

**Failure logs** 
```
>vcpkg install mongo-c-driver:x64-windows
Computing installation plan...
The following packages will be built and installed:
    mongo-c-driver[core]:x64-windows
Warning: The following VS instances are excluded because the English language pa
ck is unavailable.
    C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools
Please install the English language pack.
Detecting compiler hash for triplet x64-windows...
Starting package 1/1: mongo-c-driver:x64-windows
Building package mongo-c-driver[core]:x64-windows...
Could not locate cached archive: C:\Users\Leonid\AppData\Local\vcpkg\archives\8c
\8c1a41eb2b62b5ba43b798a4fde8f9234d9d1cf6.zip
-- Using cached C:/Users/Leonid/AppData/Local/pmm/1.4.2/vcpkg-790910f79f653978f9
0aadd958abf3c407215552/downloads/mongodb-mongo-c-driver-99d422877c5b5ea52006c13e
e3b48297251b2b2d.tar.gz
-- Extracting source C:/Users/Leonid/AppData/Local/pmm/1.4.2/vcpkg-790910f79f653
978f90aadd958abf3c407215552/downloads/mongodb-mongo-c-driver-99d422877c5b5ea5200
6c13ee3b48297251b2b2d.tar.gz
-- Applying patch fix-dependency-libbson.patch
-- Running git command: C:/Users/Leonid/AppData/Local/pmm/1.4.2/vcpkg-790910f79f
653978f90aadd958abf3c407215552/downloads/tools/git-2.26.2-1-windows/mingw32/bin/
git.exe
-- C:/Users/Leonid/AppData/Local/pmm/1.4.2/vcpkg-790910f79f653978f90aadd958abf3c
407215552/buildtrees/mongo-c-driver/patch-x64-windows-0-err.log
CMake Error at scripts/cmake/vcpkg_apply_patches.cmake:56 (message):
  Applying patch failed.  Checking patch CMakeLists.txt...

  Checking patch src/libmongoc/CMakeLists.txt...

  Checking patch src/libmongoc/build/cmake/libmongoc-1.0-config.cmake.in...

  Checking patch
  src/libmongoc/build/cmake/libmongoc-static-1.0-config.cmake.in...

  error: src/libmongoc/build/cmake/libmongoc-static-1.0-config.cmake.in:
  Filename too long

Call Stack (most recent call first):
  scripts/cmake/vcpkg_extract_source_archive_ex.cmake:141 (vcpkg_apply_patches)
  scripts/cmake/vcpkg_from_github.cmake:139 (vcpkg_extract_source_archive_ex)
  ports/mongo-c-driver/portfile.cmake:2 (vcpkg_from_github)
  scripts/ports.cmake:79 (include)


Error: Building package mongo-c-driver:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: mongo-c-driver:x64-windows
  Vcpkg version: 2020.06.15-nohash

Additionally, attach any relevant sections from the log files above.

```

--- a/scripts/cmake/vcpkg_apply_patches.cmake
+++ b/scripts/cmake/vcpkg_apply_patches.cmake
@@ -51,7 +51,7 @@ function(vcpkg_apply_patches)
         message(STATUS "Applying patch ${PATCH}")
         set(LOGNAME patch-${TARGET_TRIPLET}-${PATCHNUM})
         vcpkg_execute_in_download_mode(
-            COMMAND ${GIT} --work-tree=. --git-dir=.git apply "${ABSOLUTE_PATCH}" --ignore-whitespace --whitespace=nowarn --verbose
+            COMMAND ${GIT} -c core.longpaths=true --work-tree=. --git-dir=.git apply "${ABSOLUTE_PATCH}" --ignore-whitespace --whitespace=nowarn --verbose
             OUTPUT_FILE ${CURRENT_BUILDTREES_DIR}/${LOGNAME}-out.log
             ERROR_VARIABLE error
             WORKING_DIRECTORY ${_ap_SOURCE_PATH}

